pyccl/core.py

```diff
@@ -747,7 +758,31 @@ def compute_linear_power(self):
             rescale_mg = False
             pk = get_isitgr_pk_lin(self)
         elif trf == 'boltzmann_camb':
-            pk = get_camb_pk_lin(self)
+            pk_nl_from_camb = False
+            if self._config_init_kwargs['matter_power_spectrum'] == "camb":
+                pk_nl_from_camb = True
+            pk = get_camb_pk_lin(self, nonlin=pk_nl_from_camb)
+            if pk_nl_from_camb:
+                pk, pk_nl = pk
+                self._pk_nl['delta_matter:delta_matter'] = pk_nl
+                self._has_pk_nl = True
+                rescale_mg = False
+                rescale_s8 = False
+                if abs(self["mu_0"]) > 1e-14:
+                    warnings.warn("You want to compute the non-linear power "
+                                  "spectrum using CAMB. This cannot be "
+                                  "consistently done with mu_0 > 0.",
+                                  category=CCLWarning)
+                if np.isfinite(self["sigma8"]) \
+                        and not np.isfinite(self["A_s"]):
+                    if abs(self["sigma8"] - sigma8(self, pk)) > 1e-6:
+                        warnings.warn("You want to compute the non-linear "
```

Comment:
OK, so with this, essentially it's not really practical to use `sigma8` as an input parameter if you're going to calculate both linear and non-linear power spectra with CAMB, right?
In that case, should we throw an exception instead of a warning?

Reply:
Yeah, that might be better. I thought the pre-camb rescaling might work but I just checked and it does not.

Reply:
Yeah you'd have to call camb twice.

Reply:
Now changed to throwing an exception.